### PR TITLE
Add prettier.config setting

### PR DESCRIPTION
**Pull request by gakada**
For https://github.com/prettier/prettier-vscode/issues/254.

### CODE REVIEW

**Comment by gakada** [comment]
Did some changes. `getAbsolutePath` from `utils` is used for path handling for both ignore and config files. For `configCacheHandler`
```ts
const prettierConfigPath = getConfig().config;

if (prettierConfigPath) {
    PRETTIER_CONFIG_FILES.push(basename(prettierConfigPath));
}
```
the intention is to watch for prettier config files, for setting files restart is required, same as for `ignorePath` setting (noted in readme).

**Comment by CiGit** [comment]
Sorry for the delay.

Quickly went through (I will need a little more time to review), seems fine. 
Last question (hopefully), there is a check `hasPrettierConfig`, shouldn't it return true if a config path is given **and** that config file exists ?

**Comment by gakada** [comment]
Right, so for files like `.prettierrc` default configuration is (mostly) ignored (so it is `bundledPrettier` configuration). Added a commit for the same behavior when `config` is set.

Another thing, if `config` is set but doesn't exist, regular config files like `.prettierrc` are ignored, but that is also how [`prettier.resolveConfig`](https://prettier.io/docs/en/api.html#prettierresolveconfigfilepath-options) is.

**Comment by tvald** [comment]
I'd love to see this feature merged.
@gakada any chance to fix the conflict? (I'm happy to fix and make a new PR)
@CiGit was your question addressed above?

File: src/configCacheHandler.ts [hunk 1]
```diff
@@ -15,6 +17,12 @@ const PRETTIER_CONFIG_FILES = [
     'prettier.config.js',
 ];
 
+const prettierConfigPath = getConfig().config;
```
Comment (CiGit) [inline src/configCacheHandler.ts]: There may be multiple config files, one per workspace folder.
This is just reading one.

To do it correctly, we would need to set up file change hooks for every files. And remove them when needed. 

When a workspace folder is added, check the config and register a hook for it if needed
When a workspace folder is removed, remove the file watcher if it exists

File: src/configCacheHandler.ts [hunk 2]
```diff
@@ -15,6 +17,12 @@ const PRETTIER_CONFIG_FILES = [
...
+
+if (prettierConfigPath) {
+    PRETTIER_CONFIG_FILES.push(basename(prettierConfigPath));
```
Comment (CiGit) [inline src/configCacheHandler.ts]: It shouldn't be in this array, see previous comment.

File: src/utils.ts [hunk 3]
```diff
@@ -54,3 +55,21 @@ export function getGroup(group: string): PrettierSupportInfo['languages'] {
 function getSupportLanguages(version?: string) {
     return (require('prettier') as Prettier).getSupportInfo(version).languages;
 }
+
+/**
+ * Get absolute path for relative or absolute filePath.
+ * Relative to uri's workspace folder in relative case.
+ */
+export function getAbsolutePath(uri: Uri, filePath: string): string | undefined {
```
Comment (CiGit) [inline src/utils.ts]: :+1:

**Comment by ntotten** [comment]
Closing as this is very outdated.

**Pull request closed without merging**